authority-portal-frontend/src/app/pages/dashboard-page/dashboard-component-uptime-card/dashboard-component-uptime-card.component.ts

```diff
@@ -43,19 +43,24 @@ export class DashboardComponentUptimeCardComponent {
     this.chartData = value
       .mapNotNull((it) => this.buildDonutChartData(it))
       .orElse(null);
-    this.upSinceTooltip = value
+    this.upSinceHumanized = value
       .mapNotNull((it) =>
-        it.upSince.seconds
-          ? `Up for ${humanizeDuration(it.upSince.seconds)}`
-          : null,
+        it.upSinceSeconds ? humanizeDuration(it.upSinceSeconds) : null,
       )
       .orElse(null);
   }
   get data(): Fetched<UptimeStatusDto | undefined> {
     return this._data;
   }
   chartData: DonutChartData | null = null;
-  upSinceTooltip: string | null = null;
+  upSinceHumanized: string | null = null;
+
+  humanizeDuration(seconds: number | undefined | null): string {
+    if (seconds !== undefined && seconds !== null) {
+      return humanizeDuration(seconds);
+    }
+    return 'time period';
```

Comment:
Is this `if` necessary here? If I see this correctly, you're checking `upSinceSeconds` for `null` in `line 48`.
Also, why are you returning 'time period' here? Is this a placeholder?

Comment: Also, this function naming really confused me. I needed a minute to figure out that this is indeed not a recursive call...

Reply:
makes sense, changed the name of the function. the reason is the "hacky" import in angular of functions from another component

Reply:
makes sense, changed the name of the function. the reason is the "hacky" import in angular of functions from another component